### PR TITLE
Expose Getxxx() method for other resources

### PR DESCRIPTION
**Pull request by Lance52259**
**What this PR does / why we need it**:
Expose Getxxx() method for other resources to get vpc and subnet information.

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
```
NONE
```

**Special notes for your reviewer**:

**Release note**:

```release-note
1. move vpc and subnet to service directory.
2. expose Getxxx() method for other resources.
```

## PR Checklist

* [x] Tests added/passed.
* [ ] Documentation updated.
* [ ] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/vpc' TESTARGS='-run=TestAccVpcV1_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/vpc -v -run=TestAccVpcV1_basic -timeout 360m -parallel 4
=== RUN   TestAccVpcV1_basic
=== PAUSE TestAccVpcV1_basic
=== CONT  TestAccVpcV1_basic
--- PASS: TestAccVpcV1_basic (86.36s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/vpc       86.472s
```
```
make testacc TEST='./huaweicloud/services/acceptance/vpc' TESTARGS='-run=TestAccVpcV1_WithEpsId'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/vpc -v -run=TestAccVpcV1_WithEpsId -timeout 360m -parallel 4
=== RUN   TestAccVpcV1_WithEpsId
=== PAUSE TestAccVpcV1_WithEpsId
=== CONT  TestAccVpcV1_WithEpsId
--- PASS: TestAccVpcV1_WithEpsId (49.77s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/vpc       49.826s
```
```
make testacc TEST='./huaweicloud/services/acceptance/vpc' TESTARGS='-run=TestAccVpcV1_WithCustomRegion'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/vpc -v -run=TestAccVpcV1_WithCustomRegion -timeout 360m -parallel 4
=== RUN   TestAccVpcV1_WithCustomRegion
=== PAUSE TestAccVpcV1_WithCustomRegion
=== CONT  TestAccVpcV1_WithCustomRegion
--- PASS: TestAccVpcV1_WithCustomRegion (52.52s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/vpc       52.576s
```
```
make testacc TEST='./huaweicloud/services/acceptance/vpc' TESTARGS='-run=TestAccVpcSubnetV1_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/vpc -v -run=TestAccVpcSubnetV1_basic -timeout 360m -parallel 4
=== RUN   TestAccVpcSubnetV1_basic
=== PAUSE TestAccVpcSubnetV1_basic
=== CONT  TestAccVpcSubnetV1_basic
--- PASS: TestAccVpcSubnetV1_basic (115.30s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/vpc       115.378s
```

### CODE REVIEW

File: huaweicloud/services/vpc/resource_huaweicloud_vpc_subnet.go [hunk 1]
```diff
@@ -213,15 +214,25 @@ func resourceVpcSubnetV1Create(d *schema.ResourceData, meta interface{}) error {
 
 }
 
-func resourceVpcSubnetV1Read(d *schema.ResourceData, meta interface{}) error {
-	config := meta.(*config.Config)
-	subnetClient, err := config.NetworkingV1Client(GetRegion(d, config))
+// GetVpcSubnetFromServer is a method to obtain subnet informations through subnet ID.
+func GetVpcSubnetFromServer(d *schema.ResourceData, config *config.Config, subentId string) (*subnets.Subnet, error) {
```
Comment (ShiChangkuo) [inline huaweicloud/services/vpc/resource_huaweicloud_vpc_subnet.go]: suggest to define the method as `GetVpcSubnetByID`

File: huaweicloud/services/vpc/resource_huaweicloud_vpc.go [hunk 2]
```diff
@@ -141,15 +142,24 @@ func resourceVirtualPrivateCloudV1Create(d *schema.ResourceData, meta interface{
 	return resourceVirtualPrivateCloudV1Read(d, meta)
 }
 
-func resourceVirtualPrivateCloudV1Read(d *schema.ResourceData, meta interface{}) error {
-	config := meta.(*config.Config)
-	vpcClient, err := config.NetworkingV1Client(GetRegion(d, config))
+// GetVpcFromServer is a method to obtain vpc informations through vpc ID.
+func GetVpcFromServer(d *schema.ResourceData, config *config.Config, vpcId string) (*vpcs.Vpc, error) {
```
Comment (ShiChangkuo) [inline huaweicloud/services/vpc/resource_huaweicloud_vpc.go]: suggest to define the method as `GetVpcByID`

File: huaweicloud/services/vpc/resource_huaweicloud_vpc_subnet.go (hunk 1 above)
Comment (Lance52259) [reply huaweicloud/services/vpc/resource_huaweicloud_vpc_subnet.go]: The function name has been updated.

File: huaweicloud/services/vpc/resource_huaweicloud_vpc.go (hunk 2 above)
Comment (Lance52259) [reply huaweicloud/services/vpc/resource_huaweicloud_vpc.go]: The function name has been updated.

File: huaweicloud/config/resource_schema.go [hunk 3]
```diff
@@ -0,0 +1,11 @@
+package config
```
Comment (ShiChangkuo) [inline huaweicloud/config/resource_schema.go]: suggest to move it to common package

File: huaweicloud/services/acceptance/vpc/resource_huaweicloud_vpc_subnet_test.go [hunk 4]
```diff
@@ -155,14 +156,20 @@ func testAccCheckVpcSubnetV1Exists(n string, subnet *subnets.Subnet) resource.Te
 	}
 }
 
-func testAccVpcSubnetV1_basic(rName string) string {
+func testAccVpcSubnet_base(rName string) string {
```
Comment (chengxiangdong) [inline huaweicloud/services/acceptance/vpc/resource_huaweicloud_vpc_subnet_test.go:159]: `testAccVpcSubnet_base` is a nice base function. It makes the code more concise.
A small suggestion, except for the configuration script functions, the names of functions should comply with Camelcase rules.

File: huaweicloud/services/vpc/resource_huaweicloud_vpc.go [hunk 5]
```diff
@@ -141,28 +142,28 @@ func resourceVirtualPrivateCloudV1Create(d *schema.ResourceData, meta interface{
 	return resourceVirtualPrivateCloudV1Read(d, meta)
 }
 
-func resourceVirtualPrivateCloudV1Read(d *schema.ResourceData, meta interface{}) error {
-	config := meta.(*config.Config)
-	vpcClient, err := config.NetworkingV1Client(GetRegion(d, config))
+// GetVpcById is a method to obtain vpc informations through vpc ID.
+func GetVpcById(d *schema.ResourceData, config *config.Config, vpcId string) (*vpcs.Vpc, error) {
```
Comment (chengxiangdong) [inline huaweicloud/services/vpc/resource_huaweicloud_vpc.go]: Is this function `GetVpcById` used outside the package? If it is, it is recommended to set the parameters to `region` and `id`, as the parameters are easier to understand.

File: huaweicloud/services/vpc/resource_huaweicloud_vpc_subnet.go [hunk 6]
```diff
@@ -213,21 +215,22 @@ func resourceVpcSubnetV1Create(d *schema.ResourceData, meta interface{}) error {
 
 }
 
-func resourceVpcSubnetV1Read(d *schema.ResourceData, meta interface{}) error {
-	config := meta.(*config.Config)
-	subnetClient, err := config.NetworkingV1Client(GetRegion(d, config))
+// GetVpcSubnetById is a method to obtain subnet informations through subnet ID.
+func GetVpcSubnetById(d *schema.ResourceData, config *config.Config, subentId string) (*subnets.Subnet, error) {
+	subnetClient, err := config.NetworkingV1Client(config.GetRegion(d))
```
Comment (chengxiangdong) [inline huaweicloud/services/vpc/resource_huaweicloud_vpc_subnet.go]: same suggestion as `GetVpcById`

File: huaweicloud/services/vpc/resource_huaweicloud_vpc.go (hunk 5 above)
Comment (chengxiangdong) [reply huaweicloud/services/vpc/resource_huaweicloud_vpc.go]: sorry, there is no `id`, it is `vpcId`

File: huaweicloud/services/vpc/resource_huaweicloud_vpc_subnet.go (hunk 6 above)
Comment (Lance52259) [reply huaweicloud/services/vpc/resource_huaweicloud_vpc_subnet.go]: method call description has been updated.

File: huaweicloud/services/vpc/resource_huaweicloud_vpc.go (hunk 5 above)
Comment (Lance52259) [reply huaweicloud/services/vpc/resource_huaweicloud_vpc.go]: method call description has been updated.

File: huaweicloud/config/resource_schema.go (hunk 3 above)
Comment (Lance52259) [reply huaweicloud/config/resource_schema.go]: done.

**Pull request merged**